### PR TITLE
fix: Only archive Slack channel if created by Harald

### PR DESCRIPTION
**Pull request by SEQUOIIA**

### CODE REVIEW

File: src/Harald.Infrastructure.Slack/SlackFacade.cs [hunk 1]
```diff
@@ -26,17 +27,23 @@ public class SlackFacade : ISlackFacade
...
         private readonly IDistributedCache _cache;
         private readonly SlackOptions _options;
+        private readonly string _botUserId;
 
         public SlackFacade( HttpClient client = null, IOptions<SlackOptions> options = null, IDistributedCache cache = null)
         {
             _client = client ?? new HttpClient() { BaseAddress = new System.Uri("https://slack.com", System.UriKind.Absolute) };
             _cache = cache;
             _options = options?.Value;
+            _botUserId = Environment.GetEnvironmentVariable("SLACK_API_BOT_USER_ID") ?? "";
```
Comment (ZaradarDFDS) [inline src/Harald.Infrastructure.Slack/SlackFacade.cs]: Please use the configuration system for this. We have already enabled the Env loader. :) Either add the key you need (SLACK_API_BOT_USER_ID) as a Property to the SlackOptions or inject the IConfigration interface and use the GetValue<T>(IConfiguration, String) method to fetch your value (which will ultimately resolve to the env var).

File: src/Harald.Infrastructure.Slack/SlackFacade.cs [hunk 2]
```diff
@@ -26,17 +27,23 @@ public class SlackFacade : ISlackFacade
...
         }
 
+        public string GetBotUserId()
```
Comment (ZaradarDFDS) [inline src/Harald.Infrastructure.Slack/SlackFacade.cs]: Consider using an expression bodied member (https://davefancher.com/2014/08/25/c-6-0-expression-bodied-members/) and reserve public methods for actual operations on the subsystem (Slack API). This makes it more intuitive to work with the class which is already bloated enough.

File: src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs [hunk 3]
```diff
@@ -192,7 +192,17 @@ public async Task<IActionResult> AddConnection(ConnectionDto connection)
 
                 if (allChannelConnections.All(c => c.ClientId.ToString().Equals(clientId)))
                 {
-                    await _slackFacade.ArchiveChannel(connection.ChannelId.ToString());
+                    var channelsAll = await _slackFacade.GetConversations();
```
Comment (ZaradarDFDS) [inline src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs]: Consider combining the lambdas and using a more expressive variable name, e.g. "unregisteredChannelHandles". Technically you could probably omit the ToList() call in the end and use variable.Count() in the if statement below. The performance hit for in-memory conversion is marginal, so its not a big deal. :)

Comment (SEQUOIIA) [reply src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs]: Check e9c5937 and 1094b00

File: src/Harald.Infrastructure.Slack/SlackFacade.cs (hunk 2 above)
Comment (SEQUOIIA) [reply src/Harald.Infrastructure.Slack/SlackFacade.cs]: Check e9c5937 and 1094b00

File: src/Harald.Infrastructure.Slack/SlackFacade.cs (hunk 1 above)
Comment (SEQUOIIA) [reply src/Harald.Infrastructure.Slack/SlackFacade.cs]: Check e9c5937 and 1094b00

**Pull request merged**